lambda/simplif.ml

```diff
@@ -294,38 +292,37 @@ let simplify_exits lam =
       | Not_found -> Lstaticraise (i,ls)
       end
   | Lstaticcatch (l1,(i,[]),(Lstaticraise (_j,[]) as l2)) ->
-      Hashtbl.add subst i ([],simplif l2) ;
-      simplif l1
+      Hashtbl.add subst i ([],simplif try_depth l2) ;
+      simplif try_depth l1
   | Lstaticcatch (l1,(i,xs),l2) ->
       let {count; max_depth} = get_exit i in
       if count = 0 then
         (* Discard staticcatch: not matching exit *)
-        simplif l1
-      else if count = 1 && max_depth <= !try_depth then begin
+        simplif try_depth l1
+      else if 
+      count = 1 && max_depth <= try_depth then begin
         (* Inline handler if there is a single occurrence and it is not
            nested within an inner try..with *)
-        assert(max_depth = !try_depth);
-        Hashtbl.add subst i (xs,simplif l2);
-        simplif l1
+        assert(max_depth = try_depth);
+        Hashtbl.add subst i (xs,simplif try_depth l2);
+        simplif try_depth l1
       end else
-        Lstaticcatch (simplif l1, (i,xs), simplif l2)
+        Lstaticcatch (simplif try_depth l1, (i,xs), simplif try_depth l2)
   | Ltrywith(l1, v, l2) ->
-      incr try_depth;
-      let l1 = simplif l1 in
-      decr try_depth;
-      Ltrywith(l1, v, simplif l2)
-  | Lifthenelse(l1, l2, l3) -> Lifthenelse(simplif l1, simplif l2, simplif l3)
-  | Lsequence(l1, l2) -> Lsequence(simplif l1, simplif l2)
-  | Lwhile(l1, l2) -> Lwhile(simplif l1, simplif l2)
+      let l1 = simplif (try_depth + 1) l1 in
+      Ltrywith(l1, v, simplif try_depth l2)
+  | Lifthenelse(l1, l2, l3) -> Lifthenelse(simplif try_depth l1, simplif try_depth l2, simplif try_depth l3)
```

Comment:
In general the compiler codebase has to stay under 80 columns wide (see the `check-typo` test that you can run locally: `./tools/check-typo-since trunk` from your branch). I expect that you will have to break new lines at many places in your patch.